instrumentation/grizzly-client-1.9/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/client/ClientResponseAdvice.java

```diff
@@ -34,9 +35,12 @@
   public static Scope onEnter(
       @Advice.This final AsyncCompletionHandler<?> handler,
       @Advice.Argument(0) final Response response) {
+
+    // TODO I think all this should happen on exit, not on enter.
+    // After response was handled by user provided handler.
```

Comment:
My line of thought was the following: in comes a request, when a response is ready `AsyncCompletionHandler` will be called. I think we want to end the request processing span _after_ that handler has completed its job. So that if it fails or takes long time, that would be recorded in the captured span. No?

Reply:
i think the `CLIENT` span timing and success are supposed to match the HTTP timing / success as closely as possible

i don't see this called out in the spec, but there's some related discussion here: https://github.com/open-telemetry/opentelemetry-specification/issues/591#issuecomment-626253508